kpops/components/base_components/pipeline_component.py

```diff
@@ -124,6 +130,34 @@ def set_input_topics(self) -> None:
             for name, topic in self.from_.topics.items():
                 self.apply_from_inputs(name, topic)
 
+    def __get_all_input_topics(self) -> list[str]:
+        input_topics = self.get_input_topics()
+        extra_input_topics = self.get_extra_input_topics()
+        all_input_topics: list[str] = []
+        if input_topics is not None:
+            all_input_topics.extend(input_topics)
+        if extra_input_topics:
+            all_input_topics.extend(chain(*extra_input_topics.values()))
+        return all_input_topics
+
+    @property
+    def input_topics(self) -> Iterator[str]:
+        yield from self.__get_all_input_topics()
```

Comment:
```suggestion
    @property
    def inputs(self) -> Iterator[str]:
        yield from self.get_input_topics()
        for role_topics in self.get_extra_input_topics().values():
            yield from role_topics
```

Reply:
similar for outputs

Reply:
the problem with that is that `get_extra_input_topics` can be None and we would need to create a dict with something there as default, what not so cool is

Reply:
I would change `get_extra_input_topics` to return an empty dict